backend/src/movements.rs

```diff
@@ -78,44 +76,144 @@ impl MapModel {
                         serde_json::to_value(from_geometry).unwrap(),
                     );
                     f.set_property("to_geometry", serde_json::to_value(to_geometry).unwrap());
-                    // Editing isn't possible yet
-                    f.set_property("edited", false);
+                    f.set_property(
+                        "edited",
+                        !self.original_turn_restrictions[i.id.0].contains(&(from.id, to.id)),
+                    );
+                    f.set_property("intersection", i.id.0);
+                    f.set_property("from_road", from.id.0);
+                    f.set_property("to_road", to.id.0);
                     features.push(f);
                 }
             }
         }
 
         GeoJson::from(features)
     }
+
+    pub fn get_turn_restriction_targets(&self, from: RoadID) -> GeoJson {
+        let from = self.get_r(from);
+        let mut features = Vec::new();
+        // TODO Account for one-ways
+        for i in [from.src_i, from.dst_i] {
+            let intersection = self.get_i(i);
+            for r in &intersection.roads {
+                if *r == from.id {
+                    continue;
+                }
+                // If there's already a TR between these two, skip it.
+                if self.turn_restrictions[i.0].contains(&(from.id, *r)) {
+                    continue;
+                }
+
+                let to = self.get_r(*r);
+
+                let (abs_bearing_1, abs_bearing_2) = intersection.bearing_of_roads(from, to);
+                let kind = classify_relative_bearing(abs_bearing_1, abs_bearing_2);
+
+                let mut f = self.mercator.to_wgs84_gj(&to.linestring);
+                f.set_property("road", r.0);
+                f.set_property("kind", kind);
+                if let Some(name) = to.tags.get("name") {
+                    f.set_property("name", name.clone());
+                }
+                features.push(f);
+            }
+        }
+        GeoJson::from(features)
+    }
+
+    /// Find (intersection, from, to) to represent a turn restriction
```

Comment:
How do we robustly represent turn restrictions and make sure reasonable OSM edits don't break things? Simplest approach seems to be

1) Store a point for the intersection, and hope the closest one doesn't change much.   (This might be wrong if somebody does something like turn a single node with dual carriageways intersecting into 4 nodes, but we're kind of hopeless in that situation anyway)
2) Store the bearing of the road pointing into the intersection and then away from the intersection, calculate that bearing again, find the closest match by rotation

Reply:
This deserves some careful unit tests at some point, along with the other logic for loading savefiles

Reply:
It is a little weird, maybe, to use GJ as our output format and have multiple points in exactly the same place for representing multiple turn restrictions. We could also have one point with an array, but then what if you delete an existing TR somewhere and add a new one at the same intersection? It'd be inconsistent with our savefile format to express the "new thing" vs "deleted existing" in a different way...